flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -375,9 +376,24 @@ private int getParallelism() {
                     * effectiveConfig.get(TaskManagerOptions.NUM_TASK_SLOTS);
         }
 
+        Optional<Integer> maxOverrideParallelism = getMaxParallelismFromOverrideConfig();
+        if (maxOverrideParallelism.isPresent() && maxOverrideParallelism.get() > 0) {
+            return maxOverrideParallelism.get();
+        }
+
         return spec.getJob().getParallelism();
     }
 
+    private Optional<Integer> getMaxParallelismFromOverrideConfig() {
```

Comment:
If this method is used to determine the number of task slots, that won't work if slot sharing is disabled or if multiple slot sharing groups are used.

Reply:
You are completely right @mxm however this is a general limitation of the standalone mode itself.